src/ImageSharp/Formats/Gif/GifEncoder.cs

```diff
@@ -33,7 +33,7 @@ public sealed class GifEncoder : IImageEncoder, IGifEncoderOptions
         /// <summary>
         /// Gets or sets the color table mode: Global or local.
         /// </summary>
-        public GifColorTableMode ColorTableMode { get; set; }
+        public GifColorTableMode? ColorTableMode { get; set; }
```

Comment:
This is a breaking API change. If we plan to introduce the same logic for all formats, we should introduce this pattern in all encoders now, even if the implementation is not yet here!

Reply:
Agreed